crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -2708,12 +2718,48 @@ pub(crate) mod tests {
         assert!(!second_registration_message_mapped);
     }
 
+    #[tokio::test]
+    async fn handles_empty_service_type_2383() {
+        let tmp_dir = TempTedgeDir::new();
+        let mut config = c8y_converter_config(&tmp_dir);
+        config.service_type = String::new();
+
+        let (mut converter, _) = create_c8y_converter_from_config(config);
+
+        let service_health_message = Message::new(
+            &Topic::new_unchecked("te/device/main/service/service1/status/health"),
+            serde_json::to_string(&json!({"status": "up"})).unwrap(),
+        );
+
+        let output = converter.convert(&service_health_message).await;
+        let service_creation_message = output
+            .into_iter()
+            .find(|m| m.topic.name == SMARTREST_PUBLISH_TOPIC)
+            .expect("service creation message should be present");
+
+        let mut smartrest_fields = service_creation_message.payload_str().unwrap().split(',');
+
+        assert_eq!(smartrest_fields.next().unwrap(), "102");
+        assert_eq!(
+            smartrest_fields.next().unwrap(),
+            format!("{}:device:main:service:service1", converter.device_name)
+        );
+        assert_eq!(smartrest_fields.next().unwrap(), "service");
+        assert_eq!(smartrest_fields.next().unwrap(), "service1");
+        assert_eq!(smartrest_fields.next().unwrap(), "up");
```

Comment:
You could use that helper.
```suggestion
        assert_messages_matching(
            &messages,
            [
                (
                    SMARTREST_PUBLISH_TOPIC,
                    "102,test-device:device:main:service:service1,service,service1,up".into(),
                ),
            ],
        );
```